AEP/aep-00004.md

```diff
@@ -0,0 +1,122 @@
+Authentication would be disabled on the clusters, or at least would allow the load balancer to perform any action.
+This would greatly complexify the Load Balancer itself.
+
+Next possibility would be to let the load balancer impersonate users to the cluster using the standard impersonification faeture of the authentication system.
+This would force te load balancer to have a dedicated user on the clusters with the permission to impersonnate any user.
+As it is currently not possible to chain impersonifications, it would also forbid end-users to impersonate themselves, even if they have the permission to do it.
+
+Last possibility would be make the load balancer forward the identity headers to the clusters, in the same way the nginx ingress transmit the identity to the control plane currently.
+Extra configuration at the cluster ingress would be required to allow the load balacner to set those sensitive headers.
+This last solution is the preferred one as it is both simple to implement, secure, and featureful.
+
+### Partitions
+
+As a single session cannot span multiple clusters, all the partitions requested by the session must exist on the target cluster.
+The load-balancer would be able to select a cluster that support all the partitions requested, if any.
+It should avoid selecting a cluster that does not support a partition, and if no cluster support all the partitions, the load balancer would raise an error.
+
+As an example, consider a cluster A with partitions p1 and p2, and a cluster B with partitions p2 and p3.
+If a session need to target p2, it can go on either A or B.
+If a session targets p1 and p2, it can only go on A.
+If a session targets p3, it can only go on B.
+If a session targets p1 and p3, it cannot go anywhere, and the load balancer raises an error.
+If a session targets p4, the load balancer raises an error.
+
+### Monitoring
+
+The presence of load balancers would make the monitoring harder because of the listing accross clusters.
+
+The recommended way would be to enforce listing tasks and results on sessions, and adapt the GUI to match that requirement.
+To have a global overview, it would be necessary to add some session-wide statistics like the number of tasks completed and in error to enable the view at the session level, without the need to query all the clusters.
+The aggregation could be performed at regular interval on all opened sessions to reduce the quantity of sessions to query, as a closed session cannot have more tasks added.
+
+### Cluster field
+
+An alternative to the session id field being used to load balance the request, we could add a cluster field to all the rpc, and use that to load balance the requests.
+It could also be use for the monitoring instead of imposing a filter on the session.
+
+This change would be more intrusive API-wise.
```

Comment:
[gRPC documentation](https://grpc.io/blog/grpc-load-balancing/#l3l4-transport-vs-l7-application) suggests using cookies

Reply:
This is not about the connection, it is about the metadata of our sessions (and results/tasks).
Also, they say "cookies or similar".
And mandating the presence of a cookie is also a breaking change API wise, just not recorded in the proto definition.